examples/gno.land/r/gov/dao/dao.gno

```diff
@@ -2,27 +2,44 @@ package govdao
 
 import (
 	"std"
+	"strconv"
 
+	"gno.land/p/demo/ufmt"
 	"gno.land/p/gov/proposal"
 )
 
-var proposals = make([]Proposal, 0)
+var proposals = make([]*Proposal, 0)
 
 // XXX var members ...
 
 // Proposal represents a proposal in the governance system.
+// XXX: make it an interface with various proposal implementations
 type Proposal struct {
 	author   std.Address
 	idx      int
 	Comment  string
 	Executor proposal.Executor
+	accepted bool
+	finished bool
+}
+
+func (p Proposal) Status() string {
+	if p.Executor.Done() {
+		return p.Executor.Status()
+	}
+	if p.accepted {
+		return "accepted"
```

Comment:
What do you think about defining a type alias for these states?

ex:

```go
type Status string

var (
    Accepted Status = "accepted"
    // ...
)
```

Reply:
oh yes 👍

Reply:
fixed in 8ecee2f